content/en/docs/concepts/services-networking/topology-aware-hints.md

```diff
@@ -0,0 +1,127 @@
+
+The EndpointSlice controller is responsible for setting hints on EndpointSlices
+when this feature is enabled. The controller allocates a proportional amount of
+endpoints to each zone. This proportion is based on the number of CPU cores that
+have been allocated to nodes running in that zone. For example, if one zone had
+2 CPU cores and another zone only had 1 CPU core, the controller would allocated
+twice as many endpoints to the zone with 2 CPU cores.
+
+### Kube-Proxy
+
+Kube-Proxy filters the endpoints it routes to based on the hints set by the
+EndpointSlice controller. Essentially, the controller allocates endpoints to
+specific zones, and kube-proxy is responsible for ensuring that it only routes
+to endpoints with hints for the zone it is in.
+
+## Safeguards
+
+Both the EndpointSlice controller and Kube-Proxy implementations have a number
+of safeguards built in that will result in falling back to the original
+approach. This results in traffic being routed to all endpoints, instead of just
+a topology aware subset. Here's what could trigger that:
+
+1. **Insufficient number of endpoints:** If there are less endpoints than zones
+   in a cluster, the controller will not assign any hints. 
+
+2. **Impossible to achieve balanced allocation:** In some cases, it will be
+   impossible to achieve a balanced allocation of endpoints among zones. For
+   example, if zone-a is twice as large as zone-b, but there are only 2
+   endpoints, an endpoint allocated to zone-a may receive twice as much traffic
+   as zone-b. The controller wil not assign hints if it can't get this "expected
+   overload" value below 20% for each zone. Importantly this is not based on
+   real-time feedback. It is still possible for individual endpoints to become
+   overloaded.
+
+3. **One or more Nodes has insufficient information:** If a node does not have a
+   `topology.kubernetes.io/zone` label or is not reporting a value for
+   allocatable CPU, this functionality will be disabled for all Services in a
+   cluster.
+
+4. **One or more endpoints does not have a zone hint:** When this happens,
+   kube-proxy assumes that a transition from or to Topology Aware Hints is
+   underway. Filtering endpoints for a Service in this state would be dangerous
+   so Kube-Proxy falls back to using all endpoints.
+
+5. **A zone is not represented in hints:** If kube-proxy is unable to find at
+   least one endpoint with a hint targeting the zone it is running in, it will
+   fall back to using endpoints from all zones. This is most likely to happen as
+   a new zone is being added to a cluster.
+
+## Constraints
+
+* Topology Aware Hints are not compatible with `internalTrafficPolicy=Local`,
+  and therefore a Service cannot use both of these features. It is possible to
+  use both features in the same cluster on different Services, just not on the
+  same Service.
+
+* This approach will not work well for Services that have a large proportion of
```

Comment:
The APPROACH might, but the implementation will need to adapt to handle it, right?

Reply:
For an alpha feature, it's OK to document current shortcomings without making a commitment about future improvements to address them (we don't want to make promises unless we're sure of delivering on them).

Reply:
This is worth further discussion, but the only ways I can think to adapt to this would be new metrics that could enable feedback-based balancing or extra configuration.

Reply:
What I meant is that the implementation could evolve to consider feedback and the `hints` API would still be appropriate